### DISCUSSION TITLE
Is it possible to connect to the supabase database as an auth user with a connection string?

### DISCUSSION THREAD

**Discussion by darylhjd** [supabase/supabase#19889] [Questions]
I am using Supabase's built-in OAuth2 integration. This creates user rows in the auth.users table. 

Is there a way to set up connections to the database from my backend as a specific user this way? I am trying to use a connection string rather than using the community libraries (i.e. supabase-go).

I am aware that the library allows such a thing using the Authorization header but in this case I will not be using it.

**Comment by GaryAustin1** [comment]
If you want to use things like auth.uid() or TO role in RLS policies then you have to set that up on each session call.
```sql
set session role authenticated;
set request.jwt.claims to '{"role":"authenticated", "sub":"c8fc722a-22fb-4483-aab7-6c2be88bc03c", "app_metadata":{"role":"Student"}}';
```
Is an example I use to test running as a user in the auth table.
Somewhere I've seen some functions written to actually load the info in the settings from the auth table directly based on a user email.
